tests/nfs_create_retain_pv_disable_exporter_test.go

```diff
@@ -0,0 +1,311 @@
+
+			err = Client.createStorageClass(&storagev1.StorageClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: scName,
+					Annotations: map[string]string{
+						"openebs.io/cas-type":   "nfsrwx",
+						"cas.openebs.io/config": string(casObjStr),
+					},
+				},
+				Provisioner:   "openebs.io/nfsrwx",
+				ReclaimPolicy: &scReclaimPolicy,
+			})
+			Expect(err).To(BeNil(), "while creating SC{%s}", scName)
+		})
+	})
+
+	When("Removing volume-events-collector sidecar", func() {
+		It("should remove volume-events-collector sidecar", func() {
+			err := removeEventsCollectorSidecar(OpenEBSNamespace, nfsProvisionerName)
+			Expect(err).To(BeNil(), "while removing volume-event-exporter sidecar")
+
+		})
+	})
+
+	When("PVC with storageclass openebs-rwx is created", func() {
+		It("should create a pvc ", func() {
+			By("creating above pvc")
+			err := Client.createPVC(&corev1.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      pvcName,
+					Namespace: applicationNamespace,
+				},
+				Spec: corev1.PersistentVolumeClaimSpec{
+					StorageClassName: &scName,
+					AccessModes:      accessModes,
+					Resources: corev1.ResourceRequirements{
+						Requests: map[corev1.ResourceName]resource.Quantity{
+							corev1.ResourceStorage: resource.MustParse(capacity),
+						},
+					},
+				},
+			})
+			Expect(err).To(BeNil(), "while creating pvc %s/%s", applicationNamespace, pvcName)
+
+			pvcPhase, err := Client.waitForPVCBound(applicationNamespace, pvcName)
+			Expect(err).To(BeNil(), "while waiting for pvc %s/%s bound phase", applicationNamespace, pvcName)
+			Expect(pvcPhase).To(Equal(corev1.ClaimBound), "pvc %s/%s should be in bound phase", applicationNamespace, pvcName)
+
+			// TODO: Remove below lines after merging https://github.com/openebs/dynamic-nfs-provisioner/pull/97 PR
+			err = markNFSResources(applicationNamespace, pvcName)
+			Expect(err).To(BeNil(), "while marking for events")
+		})
+	})
+
+	When("PVC gets into bounded state", func() {
+		It("should have NFS server in running state", func() {
+			pvcObj, err := Client.getPVC(applicationNamespace, pvcName)
+			Expect(err).To(BeNil(), "while fetching pvc %s/%s", applicationNamespace, pvcName)
+			nfsPVName = pvcObj.Spec.VolumeName
+			backendPVCName = "nfs-" + nfsPVName
+
+			nfsServerLabelSelector := "openebs.io/nfs-server=nfs-" + pvcObj.Spec.VolumeName
+			err = Client.waitForPods(OpenEBSNamespace, nfsServerLabelSelector, corev1.PodRunning, 1)
+			Expect(err).To(BeNil(), "while verifying NFS Server running status")
+		})
+
+		It("should not have sent the events for PVC creation", func() {
+			var isCreateEventReceived bool
+			for retry := 5; retry >= 0; retry-- {
+				backendPVCObj, err := Client.getPVC(OpenEBSNamespace, backendPVCName)
+				Expect(err).To(BeNil(), "while fetching backend pvc %s/%s", OpenEBSNamespace, backendPVCName)
+
+				_, isCreateEventReceived = backendPVCObj.Annotations[nfs.VolumeDeleteNFSPVKey]
```

Comment:
We need to check for create event key, since we are checking for create key
```suggestion
				_, isCreateEventReceived = backendPVCObj.Annotations[nfs.VolumeCreateNFSPVKey]
```